model/simulationtests/multi_stage_electric.rb

```diff
@@ -0,0 +1,154 @@
+                     'plenum_height' => 1,
+                     'perimeter_zone_depth' => 3 })
+
+# add windows at a 40% window-to-wall ratio
+model.add_windows({ 'wwr' => 0.4,
+                    'offset' => 1,
+                    'application_type' => 'Above Floor' })
+
+always_on = model.alwaysOnDiscreteSchedule
+
+# In order to produce more consistent results between different runs,
+# we sort the zones by names
+zones = model.getThermalZones.sort_by { |z| z.name.to_s }
+
+# add thermostats
+model.add_thermostats({ 'heating_setpoint' => 20,
+                        'cooling_setpoint' => 30 })
+
+# get the heating and cooling setpoint schedule to use later
+thermostat = model.getThermostatSetpointDualSetpoints[0]
+heating_schedule = thermostat.heatingSetpointTemperatureSchedule.get
+cooling_schedule = thermostat.coolingSetpointTemperatureSchedule.get
+
+# Unitary System with CoilHeatingDXMultiSpeed, CoilCoolingDXMultiSpeed, and CoilHeatingElectricMultiStage test
+zone = zones[0]
+
+staged_thermostat = OpenStudio::Model::ZoneControlThermostatStagedDualSetpoint.new(model)
+staged_thermostat.setHeatingTemperatureSetpointSchedule(heating_schedule)
+staged_thermostat.setNumberofHeatingStages(2)
+staged_thermostat.setCoolingTemperatureSetpointBaseSchedule(cooling_schedule)
+staged_thermostat.setNumberofCoolingStages(2)
+zone.setThermostat(staged_thermostat)
+
+air_system = OpenStudio::Model::AirLoopHVAC.new(model)
+supply_outlet_node = air_system.supplyOutletNode
+
+# Modify the sizing parameters for the air system
+air_loop_sizing = air_system.sizingSystem
+air_loop_sizing.setCentralHeatingDesignSupplyAirTemperature(OpenStudio.convert(104, 'F', 'C').get)
+
+controllerOutdoorAir = OpenStudio::Model::ControllerOutdoorAir.new(model)
+outdoorAirSystem = OpenStudio::Model::AirLoopHVACOutdoorAirSystem.new(model, controllerOutdoorAir)
+outdoorAirSystem.addToNode(supply_outlet_node)
+
+fan = OpenStudio::Model::FanConstantVolume.new(model, always_on)
+heat = OpenStudio::Model::CoilHeatingDXSingleSpeed.new(model)
+heat.setName('Multi Stage DX Htg Coil')
+cool = OpenStudio::Model::CoilCoolingDXSingleSpeed.new(model)
+cool.setName('Multi Stage DX Clg Coil')
+supp_heat = OpenStudio::Model::CoilHeatingElectricMultiStage.new(model)
+supp_heat.setName('Multi Stage Sup Elec Htg Coil')
+supp_heat.setAvailabilitySchedule(always_on)
+supp_heat_stage_1 = OpenStudio::Model::CoilHeatingElectricMultiStageStageData.new(model)
+supp_heat_stage_1.setEfficiency(0.95)
+supp_heat_stage_1.setNominalCapacity(42000)
+supp_heat_stage_2 = OpenStudio::Model::CoilHeatingElectricMultiStageStageData.new(model)
+supp_heat_stage_2.setEfficiency(0.9)
+supp_heat_stage_2.setNominalCapacity(43000)
+supp_heat_stage_3 = OpenStudio::Model::CoilHeatingElectricMultiStageStageData.new(model)
+supp_heat_stage_3.setEfficiency(0.85)
+supp_heat_stage_3.setNominalCapacity(44000)
+supp_heat.addStage(supp_heat_stage_1)
+supp_heat.addStage(supp_heat_stage_2)
+supp_heat.addStage(supp_heat_stage_3)
```

Comment:
Would be good to use setStages for at least one of the two examples.

Perhaps it'd be good to also fully demonstrate the API (such as parentCoil etc...), but I don't have time to change it.